### ISSUE TITLE
[bug] chapter images not displayed even with native playback

### ISSUE THREAD

**Issue by jathek** [jellyfin/jellyfin-kodi#718]
**Describe the bug**

On Kodi, videos do not have chapter images, even though they are present in Jellyfin. The current readme says "Chapter images are missing unless native playback mode is used", but I don't see them even with native playback.

**To Reproduce**

1. Use Jellyfin for Kodi and sync with native playback
2. Play a video with chapters
3. Chapter images are blank

**Expected behavior**

Chapter images should not be blank.

**Logs**

**Screenshots**
No chapter images in Kodi:
![no_chapters](https://user-images.githubusercontent.com/692914/230510550-a10ef99a-9b01-4473-896a-2309af8be224.jpg)
Chapter images in Jellyfin:
![image](https://user-images.githubusercontent.com/692914/230510843-b51e09c9-2ec1-4aec-b1ce-116474d8f741.png)

**System (please complete the following information):**
 - OS: unRAID 6.11.5 (Slackware 15+) for Jellyfin, Nvidia Shield FW 9.1.1
 - Jellyfin Version: 10.8.9
 - Kodi Version: 20.0
 - Addon Version: 0.7.10+py3
 - Playback Mode: Native

**Additional context**

**Comment by DaisyF8** [comment]
It's just a setting.

I also use native mode. In my skin, Arctic Horizon 2, I need to enable Settings/Media/Videos/ExtractChapterThumbnails and then they are there!

Good luck

**Comment by jathek** [comment]
> It's just a setting.
> 
> 
> 
> I also use native mode. In my skin, Arctic Horizon 2, I need to enable Settings/Media/Videos/ExtractChapterThumbnails and then they are there!
> 
> 
> 
> Good luck

Yes, I avoided enabling by that because I thought Jellyfin's chapter images could be used somehow. I think enabling that setting would just cause Kodi to extract its own images?

**Comment by DaisyF8** [comment]
I can't say for sure if Kodi is using its own extraction or using Jellyfins chapter images, probably its own.

But it will do the extracting only for videos which have defined chapters. So maybe some/most movies and very little tv-episodes (do you know of any tv?). The extraction of chapter images  doesn't take long. And on the network Kodi won't need to stream the entire movie, it just seeks to those times.  For example ffmpeg takes a few second for 10 frames.

So just test it, it will take less time than you imagine.

What may add lots of time is IF you enable Jellyfin to generate chapters (every 5 minutes) when there are no chapters. Because this will be applied to all your media.  Jellyfin will do this for every video. And IF Jellyfin then tells Kodi "I got chapters for EVERYTHING (evil laugh)" the harddrive-light will keep going disco fever into the night because Kodi does it a second time. Bonus points if Kodi and Jellyfin do this at the same time.

But U don't think the second IF can be true. But I haven't tested if Jellyfinhgenerated chapters get forwarded to Kodi.

**Comment by DaisyF8** [comment]
By the way awesome 👍 that you replied to me when the question was 2 years old.  I replied for others having the same problem but I hope I helped you.

**Comment by oddstr13** [comment]
Chapter images is a *relatively* new feature in Jellyfin, and has not yet been implemented in this Kodi addon.

**Comment by jathek** [comment]
> Chapter images is a _relatively_ new feature in Jellyfin, and has not yet been implemented in this Kodi addon.
> 
> Edit: as for native playback, none of that is done by the addon, so if if that doesn't work, I don't think there is much for us to fix.

I switched to addon mode about a year ago and chapter images don't work that way either, although from what you said, it seems like that's expected since it's not implemented. I hope it will one day! My library is huge and I would rather not have Kodi scrape it all for chapter images that have already been scraped elsewhere.

**Comment by oddstr13** [comment]
Yep, no chapter images is to be expected with addon mode.

Contributions toward getting it implemented, be it research or code, but as I just commented on the extras feature request, I don't have time to dedicate towards this at the moment.

**Comment by mcarlton00** [comment]
Pretty sure most of the code is already there for it.  We just need to generate image URLs and add them to the database (and hope that it can use URLs for chapter images instead of just local images).

* https://kodi.wiki/view/Databases/MyVideos#bookmark
* https://github.com/jellyfin/jellyfin-kodi/blob/17f3053613e110851ffa96687a05e406640f0f1e/jellyfin_kodi/objects/kodi/queries.py#L253-L265

**Comment by youhitgit** [comment]
Somebody called @TeamB working on the embycon plugin got this working. He talks about it in post #16 on this page.

https://emby.media/community/index.php?/topic/116095-embycon-and-chapter-thumbnails-possible-within-kodi/

Hopefully this helps
